### DISCUSSION TITLE
Do Cirrus's Linux builds use seccomp?

### DISCUSSION THREAD

**Discussion by asomers** [cirruslabs/cirrus-ci-docs#1182] [Q&A]
Sometime within the last 2 weeks tests for my project (https://github.com/nix-rust/nix) began failing because [personality](https://man7.org/linux/man-pages/man2/personality.2.html) now returns `EPERM`.  Google suggests that this might be caused by certain seccomp profiles.  But there's nothing in Cirrus's docs about seccomp.  Is it turned on?  How can I find out?

**Comment by fkorotkov** [comment]
We do run some parts of x86 Linux containers and all Arm Linux containers in GKE Autopilot as of Thursday June 22nd 2023. GKE Autopilot [applies default containerd setcomp profile](https://cloud.google.com/kubernetes-engine/docs/concepts/seccomp-in-gke#what-is-seccomp). You can try to run your workloads in [privileged containers](https://cirrus-ci.org/guide/linux/#kvm-enabled-privileged-containers) which are executed in dedicated virtual machines.

**Reply by asomers** [reply]
Oh, I forgot to mention that this failure is seen only on aarch64 Linux.  So that's telling.  However, adding `kvm: true` to the task doesn't fix this problem.  Maybe I'll just permanently disable that particular test in CI, now that I know that seccomp is the culprit.
